modules/installation-cloudformation-subnet-localzone.adoc

```diff
@@ -81,9 +113,19 @@ Resources:
       VpcId: !Ref VpcId
       CidrBlock: !Ref PrivateSubnetCidr
       AvailabilityZone: !Ref ZoneName
+ifdef::outposts[]
+      OutpostArn: !If [ OutpostEnabled, !Ref OutpostArn, !Ref "AWS::NoValue"]
+endif::outposts[]
       Tags:
       - Key: Name
+ifndef::outposts[]
         Value: !Join ['-', [!Ref ClusterName, "private", !Ref ZoneName]]
+endif::outposts[]
+ifdef::outposts[]
+        Value: !Join ['-', [!Ref ClusterName, !Ref PrivateSubnetLabel, !Ref ZoneName]]
+      - Key: kubernetes.io/cluster/unmanaged
```

Comment:
@JoelSpeed @jeana-redhat confirmed CCM didn't tried to add subnets with this tag. Test 1 is using the tag, test 2 without.

- Test 1) with tags:
Service LB created without annotations:

```yaml
$ oc get svc -n ${APP_NAME} myapp-outpost-clb -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2024-02-22T06:00:04Z"
  finalizers:
  - service.kubernetes.io/load-balancer-cleanup
  labels:
    app: myapp-outpost
  name: myapp-outpost-clb
  namespace: myapp-outpost
  resourceVersion: "36663"
  uid: 1322e663-06d4-4c3b-b961-98b34b25b055
spec:
  allocateLoadBalancerNodePorts: true
  clusterIP: 172.30.67.255
  clusterIPs:
  - 172.30.67.255
  externalTrafficPolicy: Cluster
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: http
    nodePort: 30400
    port: 80
    protocol: TCP
    targetPort: 8081
  selector:
    app: myapp-outpost
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - hostname: a1322e66306d44c3bb96198b34b25b05-131144436.us-east-1.elb.amazonaws.com
```

![Screenshot from 2024-02-22 03-24-22](https://github.com/openshift/openshift-docs/assets/3216894/8d47b7ca-969f-43ea-9567-170dd0019be2)


- Test 2) when removing the tags manually from the subnet and add a new svc lb, it got stuck:

```
Events:
  Type     Reason                  Age   From                Message
  ----     ------                  ----  ----                -------
  Warning  SyncLoadBalancerFailed  62s   service-controller  Error syncing load balancer: failed to ensure load balancer: ValidationError: You cannot use Outposts subnets for load balancers of type 'classic'
           status code: 400, request id: 41905aaa-da5d-4b48-8d54-50d0c6ba1be4
  Warning  SyncLoadBalancerFailed  56s  service-controller  Error syncing load balancer: failed to ensure load balancer: ValidationError: You cannot use Outposts subnets for load balancers of type 'classic'
           status code: 400, request id: fa479b53-783e-4e06-9ee3-e5d03efc254a
  Warning  SyncLoadBalancerFailed  46s  service-controller  Error syncing load balancer: failed to ensure load balancer: ValidationError: You cannot use Outposts subnets for load balancers of type 'classic'
           status code: 400, request id: 914e4c9d-02f3-4367-907a-f891675a1817
  Normal   EnsuringLoadBalancer    26s (x4 over 64s)  service-controller  Ensuring load balancer
  Warning  SyncLoadBalancerFailed  25s                service-controller  Error syncing load balancer: failed to ensure load balancer: ValidationError: You cannot use Outposts subnets for load balancers of type 'classic'
           status code: 400, request id: 5068eb6b-f63e-42d2-b27e-8ff20d18d80d
```

Reply:
+ @alebedev87 could you please validate this scenario?

Reply:
@alebedev87 confirmed on Slack:

@JoelSpeed @jeana-redhat @alebedev87  >

I just run manual tests with the following environment:
- A cluster deployed using all zones from the Region (including A which OP rack is attached)
- Subnet on OP created with "unmanaged" tag using the CloudFormation template
- ALBO created, and ingress using the public subnet created by CloudFormation template

Result: the "unmanaged" subnet can't impact ALBO/ALBC for creating ingress when it is annotating correctly with the Outpost subnet IDs.

## Log:

Check list:

1. Subnets created by CloudFormation
```sh
$ aws cloudformation  describe-stacks --stack-name ${CLUSTER_ID}-subnets-outpost | jq '.Stacks[].Outputs[] | select(.OutputKey=="PublicSubnetId")'
{
  "OutputKey": "PublicSubnetId",
  "OutputValue": "subnet-0f2c7cb846fd7c55a",
  "Description": "Subnet ID of the public subnets."
}

PUBLIC_SUBNET=$(aws cloudformation  describe-stacks --stack-name ${CLUSTER_ID}-subnets-outpost | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="PublicSubnetId").OutputValue')
```

2. Tags from those subnets
```sh
$ aws ec2 describe-subnets --subnet-ids ${PUBLIC_SUBNET} | jq '.Subnets[] | [.SubnetId, (.Tags[] | select(.Key | contains("kubernetes"))) ]'
[
  "subnet-0f2c7cb846fd7c55a",
  {
    "Key": "kubernetes.io/cluster/unmanaged",
    "Value": "true"
  }
]
```

3. Ingress created and ALB FQDN
```sh
$ oc get ingress -n ${APP_NAME}
NAME                 CLASS   HOSTS   ADDRESS                                                                   PORTS   AGE
alb-outpost-public   cloud   *       k8s-myappout-alboutpo-091a2e4eac-1527379645.us-east-1.elb.amazonaws.com   80      15s

ALB_FQDN=$(oc get ingress -n ${APP_NAME} alb-outpost-public \
  --template='{{(index .status.loadBalancer.ingress 0).hostname}}')

echo $ALB_FQDN

$ echo $ALB_FQDN
k8s-myappout-alboutpo-091a2e4eac-1527379645.us-east-1.elb.amazonaws.com
```

4. ALB subnets (created by ALBC)
```sh
$ aws elbv2 describe-load-balancers | jq -r ".LoadBalancers[] | select(.DNSName==\"${ALB_FQDN}\").AvailabilityZones"
[
  {
    "ZoneName": "us-east-1a",
    "SubnetId": "subnet-0f2c7cb846fd7c55a",
    "OutpostId": "op-0a759e7fbceffcb86",
    "LoadBalancerAddresses": []
  }
]
```

5. ALB's target instances (in the target group)
```sh
ALB_ARN=$(aws elbv2 describe-load-balancers | jq -r ".LoadBalancers[] | select(.DNSName==\"${ALB_FQDN}\").LoadBalancerArn")
TG_ARN=$(aws elbv2 describe-target-groups --load-balancer-arn $ALB_ARN  | jq -r .TargetGroups[0].TargetGroupArn)

$ aws elbv2 describe-target-health  --target-group-arn $TG_ARN
{
    "TargetHealthDescriptions": [
        {
            "Target": {
                "Id": "i-09ea38f1672bbd7be",
                "Port": 30474
            },
            "HealthCheckPort": "30474",
            "TargetHealth": {
                "State": "healthy"
            }
        }
    ]
}
```

6. Testing
```sh
$ APP_INGRESS=$(./oc get ingress -n ${APP_NAME} alb-outpost-public \
  --template='{{(index .status.loadBalancer.ingress 0).hostname}}')

while ! curl $APP_INGRESS; do sleep 5; done
GET / HTTP/1.1
X-Forwarded-For: 179.193.53.233
X-Forwarded-Proto: http
X-Forwarded-Port: 80
Host: k8s-myappout-alboutpo-091a2e4eac-1527379645.us-east-1.elb.amazonaws.com
X-Amzn-Trace-Id: Root=1-65dceb86-05ffe6523dc625cf64c07ffe
User-Agent: curl/8.0.1
Accept: */*
```

Reply:
Ingress manifest
```
$ oc get ingress alb-outpost-public -n ${APP_NAME} -o yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/subnets: subnet-0f2c7cb846fd7c55a
    alb.ingress.kubernetes.io/target-node-labels: location=outposts
    alb.ingress.kubernetes.io/target-type: instance
  creationTimestamp: "2024-02-26T19:33:27Z"
  finalizers:
  - ingress.k8s.aws/resources
  generation: 1
  labels:
    location: outpost
  name: alb-outpost-public
  namespace: myapp-outpost
  resourceVersion: "78475"
  uid: 81c74347-565a-42d2-b3b7-04543496238a
spec:
  ingressClassName: cloud
  rules:
  - http:
      paths:
      - backend:
          service:
            name: myapp-outpost-alb
            port:
              number: 80
        path: /
        pathType: Prefix
status:
  loadBalancer:
    ingress:
    - hostname: k8s-myappout-alboutpo-091a2e4eac-1527379645.us-east-1.elb.amazonaws.com
```

Reply:
@mtulio : as discussed the cluster tag is used for the auto discovery which is a different flow from the explicit subnet selection via the annotation. Thanks for confirming this with the test!